sdk/eventhub/azure-eventhub-checkpointstoretable/azure/eventhub/extensions/checkpointstoretable/_tablestoragecs.py

```diff
@@ -186,6 +201,7 @@ def _claim_one_partition(self, ownership, **kwargs):
             )
             raise OwnershipLostError()
         except Exception as error:  # pylint:disable=broad-except
+            # includes ResourceModifiedError (no matching `etag`)
```

Comment:
now I'm a bit confused after seeing the comment..

since we decide to get the `IfNotModified/eTag` back, my understanding is that we need to raise OwnershipLostError when meeting with the `ResourceModifiedError` instead of keeping the ownership.

so this comment should not be here?

Reply:
I thought that this ResourceModifiedError should be caught in this general exception rather than raising an OwnershipLostError(), b/c in JS, they don't raise an error. They just continue: https://github.com/Azure/azure-sdk-for-js/blob/7d82d3331de2dbd8db193f49bb783ae60c4a4fb9/sdk/eventhub/eventhubs-checkpointstore-table/src/tableCheckpointStore.ts#L192

thoughts on whether it should be caught here or added to above and raise an OwnershipLostError()?

Reply:
See my comment above, I think the implementation in JS one is a bit different -- they're trying to claim an array of partitions and if there's a modified error in the JS code, they don't push the partition to an array, however, here the func we're implementing is to claim a single partition.